common/src/main/java/io/druid/math/expr/Expr.java

```diff
@@ -527,7 +528,7 @@ protected final long evalLong(long left, long right)
   @Override
   protected final double evalDouble(double left, double right)
   {
-    return Evals.asDouble(left < right);
+    return Evals.asDouble(Double.compare(left, right) < 0);
```

Comment:
Could you please comment on this and similar places, not to amuse later readers why such "long" form is chosen over "simple" `left < right`.